code/modules/mob/living/carbon/brain/death.dm

```diff
@@ -1,12 +1,14 @@
 /mob/living/carbon/brain/death(gibbed)
 	if(stat == DEAD)	return
 	if(!gibbed && container && istype(container, /obj/item/device/mmi))//If not gibbed but in a container.
-		for(var/mob/O in viewers(container, null))
-			O.show_message(text("<span class='danger'>[]'s MMI flatlines!</span>", src), 1, "<span class='warning'>You hear something flatline.</span>", 2)
-		if(!istype(container, /obj/item/device/mmi/posibrain))
+		var/obj/item/device/mmi/posibrain/contained = src.container
+		if(!istype(contained))

```

Comment:
This is not how I would do this.
It's not exactly sane.

Why not do container.OnMobDeath(src) instead?